### ISSUE TITLE
Many-to-one classification

### ISSUE THREAD

**Issue by PiranjaF** [Russell91/LSTMSummation#2]
I'm already an avid user of Caffe and Apollo seems like a really good way of expanding the functionality! One thing I cannot find when going through your tutorials is whether Apollo currently has the functionality to perform 'many-to-one' classification (see http://karpathy.github.io/2015/05/21/rnn-effectiveness/) used for e.g. sentiment analysis. How would that be implemented using Apollo?

**Comment by Russell91** [comment]
This is a great question! If you think about it, the LSTMSummation example contained in this repo is an example of a many-to-one network. We input many numbers, and we output one sum.

To extend the example to a sentiment classification network, we need a few more tools though.
1) We need wordvec input layers
2) We need to use softmax and cross-entropy loss for classification, rather than the euclidean loss herein
3) We need a way to deal with the fact that separate instances within a batch may have different length.

Fortunately, you can find the answers to all of these questions in the [Machine Translation](http://github.com/russell91/mt.caffe) repository. In fact, if you take that code and simply make the target translation equal to the class you are trying to predict, you should get a sentiment analyzer! If you invest some time and go down this route, and you end up having specific problems, feel free to open an issue.

**Comment by PiranjaF** [comment]
Thanks for the quick answer, that sounds great! I'll look into it.
